eclipse-theia/theia
# Implement VS Code tree view checkbox API.

#### What it does
Implements the ability to have check boxes on tree views in plugins

Fixes #12695

Contributed on behalf of STMicroelectronics

#### How to test
Install the attached VS Code extension. It contributes a a view called "Checkbox API Test". Open the view if necessary. Play around with checking/unchecking the checkboxes. The system automatically checks/unchecks parent and child checkboxes: see the VS Code doc linked in the issue for details.
The extension also contributes a preference "Manage Checkbox State Manually". When this preference is on, the automatic checking/unchecking is disabled. The extensions writes to the log, when it is notified of checkbox state changes. There is also a context menu action to check/uncheck the tree items programmatically. 

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
